Code/Framework/AzToolsFramework/AzToolsFramework/UI/EditorEntityUi/EditorEntityUiSystemComponent.h

```diff
@@ -45,6 +46,7 @@ namespace AzToolsFramework
 
         private:
             EditorEntityUiHandlerId m_idCounter = 1;
+            AZStd::unique_ptr<EditorEntityUiHandlerBase> m_defaultEntityUiHandler;
```

Comment:
Decided to rely on a default handler as the behavior is meant to be common to vast majority of cases. To register every entity not part of another handler in this "default" handler would introduce annoying logic and likely performance bottlenecks on huge levels . The default behavior can be opt-out when registering an handler, as done for the level root.